tests/clickhouse/test_reader.py

```diff
@@ -0,0 +1,60 @@
+from typing import Sequence, Tuple
+
+import pytest
+
+from snuba import settings
+from snuba.clickhouse.errors import ClickhouseError
+from snuba.clickhouse.http import HTTPReader
+from snuba.clickhouse.native import NativeDriverReader
+from snuba.clickhouse.query import ClickhouseQuery
+from snuba.environment import clickhouse_ro
+from snuba.reader import Reader
+
+
+class SimpleClickhouseQuery(ClickhouseQuery):
+    def __init__(self, columns: Sequence[Tuple[str, str]]) -> None:
+        self.__columns = columns
+
+    def _format_query_impl(self) -> str:
+        columns = ", ".join(f"{value} as {alias}" for alias, value in self.__columns)
+        return f"SELECT {columns}"
```

Comment:
I wanted to use the AST for this but that would have meant either

1. relying on the other steps of query processing and executing the query against an actual dataset, coupling this test to a lot of unnecessary things
2. refactoring the `AstClickhouseQuery` constructor so that it would be more like a dataclass, independent of the `Query` object entirely

Both of those are a rabbit holes I have no interest in falling down into right now. We should do the second one eventually.

Reply:
To me it is fine not to use the AST to write a unit test at this level, anyway if you want to use it, you don't need to depend on dataset to build a `Query` object and transform it into a Clickhosue query since the `Query` object depends on a DataSource (which is the abstract class above TableStorage and can be created from scratch without all the heavy weight of the dataset.

Example:
If you just want a Query object starting from the AST you can do it this way
https://github.com/getsentry/snuba/blob/master/tests/query/parser/test_query.py#L20-L32
(you have to use ASTClickhouseQuery instead of DictClickhouseQuery).